mlflow/cli.py

```diff
@@ -554,6 +564,24 @@ def gc(older_than, backend_store_uri, run_ids):
     else:
         run_ids = run_ids.split(",")
 
+    deleted_experiment_ids = []
+    current_time = int(time.time() * 1000)
+    for experiment in mlflow.search_experiments(view_type=ViewType.DELETED_ONLY):
+        if older_than is None:
+            deleted_experiment_ids.append(experiment.experiment_id)
+        elif (
+            experiment.last_update_time is not None
+            and experiment.last_update_time <= current_time - time_delta
+        ):
+            deleted_experiment_ids.append(experiment.experiment_id)
```

Comment:
I didn't notice we have `backend_store`. Can we use it?

Reply:
Sorry, I didn't get it exactly. Do you mean `backend_store` to search experiments?

Reply:
```python
deleted_experiments = []
next_page_token = None
while True:
    experiments = backend_store.search_experiments(
        view_type=ViewType.DELETED_ONLY,
        page_token=next_page_token,
    )
    deleted_experiments.extend(experiments)
    if page_results.next_page_token is None:
        break
    next_page_token = page_results.next_page_token
```

Reply:
@harupy Thanks for sharing! I will update. 

Can I use `backend_store` for `search_runs` as well (Any example usage of the same)?

Reply:
Yes, we should do the same for `search_runs`. A helper function like this might be helpful to handle pagination.

```python
def paginate(func, page_token=None, **kwargs):
    page = func(page_token=page_token, **kwargs)
    yield from page
    if page.token:
        yield from paginate(func, page.token, **kwargs)


all_experiments = list(
    paginate(backend_store.search_experiments, view_type=ViewType.DELELTED_ONLY)
)
```